### PR TITLE
WotW - Shinjo Trailblazer

### LINKED ISSUE #2651: CotE: Shinjo Trailblazer
Shinjo Trailblazer

### PR DESCRIPTION
**Pull request by jamesoddy**
Closes #2651

### CODE REVIEW

File: server/game/cards/07-WotW/ShinjoTrailblazer.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const DrawCard = require('../../drawcard.js');
+
+class ShinjoTrailblazer extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            title: 'Gain +2/+2',
+            when: {
+                onCardRevealed: (event, context) => event.card.isProvince && event.card.controller === context.source.controller.opponent && this.game.isDuringConflict()
```
Comment (jamesoddy) [inline server/game/cards/07-WotW/ShinjoTrailblazer.js]: Please confirm if this is the best way to check if it's an opponent's province.  Seemed a bit wordy to me.

File: server/game/cards/07-WotW/ShinjoTrailblazer.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            },
+            gameAction: ability.actions.cardLastingEffect({ effect: ability.effects.modifyBothSkills(2) }),
+            effect: 'to give {0} +2{1}, +2{2}',
+            effectArgs: () => ['military', 'political']
```
Comment (jeremylarner) [inline server/game/cards/07-WotW/ShinjoTrailblazer.js]: You don't need to do this any more - the new chat log looks for military and political within string text

File: server/game/cards/07-WotW/ShinjoTrailblazer.js (hunk 1 above)
Comment (jeremylarner) [reply server/game/cards/07-WotW/ShinjoTrailblazer.js]: I think that's the best way to do it....

File: server/game/cards/07-WotW/ShinjoTrailblazer.js (hunk 2 above)
Comment (jamesoddy) [reply server/game/cards/07-WotW/ShinjoTrailblazer.js]: So just to clarify we would just have

```
effect: 'to give {0} +2military, +2political',
```
with no spaces?

Comment (jeremylarner) [reply server/game/cards/07-WotW/ShinjoTrailblazer.js]: Huh, you're right I hadn't thought of that.  It will  only replace them as separate words (i.e. with spaces on both sides...)
